test/langtools/tools/javac/doctree/ReferenceTest.java

```diff
@@ -83,6 +84,31 @@
  * {@link #trees Field}
  * {@link #getSupportedSourceVersion Method}
  * {@link #init(ProcessingEnvironment Method}
+ * {@link double Class}
+ * {@link double.NAN Bad}
+ * {@link double#NAN Bad}
+ * {@link double#double Bad}
+ * {@link java.base/double Bad}
+ *
+ * {@link List<String> Interface}
+ * {@link List<String>.add Bad}
+ * {@link List<String>#add Method}
+ * {@link List<String>#add(Object) Method}
+ * {@link Map<String, String>.Entry Interface}
+ * {@link Map<String, String>.Entry<String, String> Interface}
+ * {@link Map<String, String>.Entry<String, String>.getKey Bad}
+ * {@link Map<String, String>.Entry<String, String>#getKey Method}
+ * {@link Map<String, String>.Entry<String, String>#setValue(Object) Method}
+ *
+ * {@link java.base/java.util.List<String> Bad}
+ * {@link java.base/java.util.List<String>.add Bad}
+ * {@link java.base/java.util.List<String>#add Bad}
+ * {@link java.base/java.util.List<String>#add(Object) Bad}
+ * {@link java.base/java.util.Map<String, String>.Entry Bad}
+ * {@link java.base/java.util.Map<String, String>.Entry<String, String> Bad}
+ * {@link java.base/java.util.Map<String, String>.Entry<String, String>.getKey Bad}
+ * {@link java.base/java.util.Map<String, String>.Entry<String, String>#getKey Bad}
+ * {@link java.base/java.util.Map<String, String>.Entry<String, String>#setValue(Object) Bad}
```

Comment:
Is there a reason parameterized types don't work with modules, or is it just an implementation quirk? Not that it matters much as it is a rather unlikely corner case.